spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/storage/GcpStorageAutoConfiguration.java

```diff
@@ -57,11 +58,14 @@
 	public static Storage storage(CredentialsProvider credentialsProvider,
 			GcpStorageProperties gcpStorageProperties,
 			GcpProjectIdProvider projectIdProvider) throws IOException {
+		Credentials propertiesCredentials = gcpStorageProperties.getCredentials();
 		return StorageOptions.newBuilder()
-				.setCredentials(gcpStorageProperties.getCredentials().getLocation() != null
+				.setCredentials(propertiesCredentials.getLocation() != null
+						|| propertiesCredentials.getEncodedKey() != null
 						? new DefaultCredentialsProvider(gcpStorageProperties).getCredentials()
 						: credentialsProvider.getCredentials())
```

Comment:
I think this logic must be changed as well.
The user provided `CredentialsProvider` must have a precedence independently of the properties provided.
Therefore would be better to refactor this auto-configuration to extract a `CredentialsProvider` conditional bean population.

Different story as well though.